jdsol-spec/src/main/asciidoc/jdsol.adoc

```diff
@@ -1,6 +1,1240 @@
+elements:
+
+* A source path
+* The package name converted to a directory path
+* The source file name from a <<TermJDI,JDI>> call (derived from the `SourceFile`
+  class file attribute)
+
+Since existing debuggers use this mechanism (the only way for an existing
+debugger to find <<TermTranslatedSource,translated-source>>) each aspect must be
+addressed:
+
+* The source path must be set-up to include translated-source directories
+* Source must be placed in a directory corresponding to the package
+* The <<TermJDI,JDI>> call must return the translated-source name.
+
+For debuggers written against the new APIs, a new method has been added which
+returns the source path - this makes the translated-source directory structure
+flexible.
+
+=== Multiple Source Files per Class File
+
+When an inclusion mechanism is used, a class file will contain source from
+multiple <<TermTranslatedSource,translated-source>> files. The SourceFile
+attribute of class files only associates one source file with a class file which
+is one reason the approach of simply rewriting the `SourceFile` and
+`LineNumberTable` attributes had to be abandoned. The SMAP allows a virtually
+unlimited number of source files per stratum.
+
+== Source Map Format
+
+A Source Map (SMAP) describes a mapping between source positions in an input
+language (<<TermTranslatedSource,translated-source>>) and source positions in a
+generated output language. A view of the source through such a mapping is called
+a <<TermStratum,stratum>>. The <<TermSourceMapFile,SMAP-file>> contains an
+unresolved SMAP. The <<TermSourceDebugExtension,SourceDebugExtension>> class
+file attribute, when used as described in this document, contains an SMAP. The
+SMAP stored in a SourceDebugExtension attribute must be
+<<SMAP Resolution,resolved>>, and thus will have no
+<<Embedded Source Maps,embedded SMAPs>> and will have the
+<<TermFinalSource,final-source>> language as the output language.
+
+An SMAP consists of a header and one or more sections of mapping information.
+
+There are currently seven types of section: stratum sections, file sections,
+line sections, vendor sections, end sections, and open and close embedded
+sections. New section types may be added in the future - to facilitate this, any
+unknown sections must be ignored without error.
+
+The semantics of each section is discussed below. For clarity, an informal
+description of the syntax of each section is included in the discussion. See the
+formal <<SMAP Syntax,SMAP syntax>> for syntax questions.
+
+=== General Format
+
+The SMAP consists of lines of
+http://www.unicode.org/unicode/standard/standard.html[Unicode] text, with a
+concrete representation of http://ietf.org/rfc/rfc2279.txt[UTF-8]. Line
+termination is with line-feed, carriage-return or carriage-return followed by
+line-feed. Because SMAPs are included in class files, size of the SMAP was an
+important constraint on the format chosen for them.
+
+=== Header
```

Comment:
Not sure how to correct this, but `header` is a reserved word in asciidoc.

This is easiest to see in the html generated page and the odd css behavior due to this reserved word, it causes alternate formatting for this specific section heading.

It renders as ...

``` html
<h3 id="header"><a class="anchor" href="#header"></a>5.2. Header</h3>
```

The `id="header"` portion is what messes with formatting.

Reply:
Fixed with a custom ID